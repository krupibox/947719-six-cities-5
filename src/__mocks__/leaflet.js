const Leaflet = jest.genMockFromModule(`leaflet`);

Leaflet.icon = () => {};

Leaflet.map = () => {
  return {
    setView: () => {},
  };
};

Leaflet.marker = () => {
  return {
    addTo: () => {},
  };
};

Leaflet.tileLayer = () => {
  return {
    addTo: () => {},
  };
};

Leaflet.layerGroup = () => {
  return {
    addTo: () => {},
  };
};

module.exports = Leaflet;
