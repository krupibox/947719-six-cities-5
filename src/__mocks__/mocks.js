export const TestMock = {
  activeCity: `Cologne`,

  cities: [`Paris`, `Brussels`, `Hamburg`, `Amsterdam`, `Cologne`],

  firstCity: [`Cologne`],

  review: {
    user: {
      id: 1,
      avatarUrl: `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/7.jpg`,
      name: `Max`,
    },
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    date: `2020-11-05T09:55:08.500Z`,
    rating: 4.1,
  },

  reviews: [{
    user: {
      id: 1,
      avatarUrl: `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/7.jpg`,
      name: `Max`,
    },
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    date: `2020-11-05T09:55:08.500Z`,
    rating: 4.1,
  }],

  favorites: {
    Cologne: [{
      id: 1,
      city: {
        name: `Paris`,
        location: {
          latitude: 0,
          longitude: 0,
        }
      },
      previewImage: `img/apartment-01.jpg`,
      images: [
        `img/apartment-01.jpg`,
        `img/apartment-02.jpg`,
        `img/apartment-03.jpg`,
      ],
      type: `Apartment`,
      price: 80,
      rating: 3.1,
      title: `Beautiful &amp; luxurious apartment at great location`,
      coordinates: [52.3909553943508, 4.85309666406198],
      host: {
        id: 1,
        name: `Angelina`,
        isPro: true,
        avatarUrl: `img/avatar-angelina.jpg`
      },
      description: `Discover daily local life in city center.`,
      isPremium: true,
      location: {
        latitude: 0,
        longitude: 0,
      }
    }],
    Hamburg: [{
      id: 1,
      city: {
        name: `Paris`,
        location: {
          latitude: 0,
          longitude: 0,
        }
      },
      previewImage: `img/apartment-01.jpg`,
      images: [
        `img/apartment-01.jpg`,
        `img/apartment-02.jpg`,
        `img/apartment-03.jpg`,
      ],
      type: `Apartment`,
      price: 80,
      rating: 3.1,
      title: `Beautiful &amp; luxurious apartment at great location`,
      coordinates: [52.3909553943508, 4.85309666406198],
      host: {
        id: 1,
        name: `Angelina`,
        isPro: true,
        avatarUrl: `img/avatar-angelina.jpg`
      },
      description: `Discover daily local life in city center.`,
      isPremium: true,
      location: {
        latitude: 0,
        longitude: 0,
      }
    }],
  },

  offer: {
    id: 1,
    city: {
      name: `Paris`,
      location: {
        latitude: 0,
        longitude: 0,
      }
    },
    previewImage: `img/apartment-01.jpg`,
    images: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
    ],
    type: `Apartment`,
    price: 80,
    rating: 3.1,
    title: `Beautiful &amp; luxurious apartment at great location`,
    coordinates: [52.3909553943508, 4.85309666406198],
    host: {
      id: 1,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `Discover daily local life in city center.`,
    isPremium: true,
    isFavorite: true,
    location: {
      latitude: 0,
      longitude: 0,
    }
  },

  offers: [
    {
      id: 1,
      city: {
        name: `Paris`,
        location: {
          latitude: 0,
          longitude: 0,
        }
      },
      previewImage: `img/apartment-01.jpg`,
      images: [
        `img/apartment-01.jpg`,
        `img/apartment-02.jpg`,
        `img/apartment-03.jpg`,
      ],
      type: `Apartment`,
      price: 80,
      rating: 3.1,
      title: `Beautiful &amp; luxurious apartment at great location`,
      coordinates: [52.3909553943508, 4.85309666406198],
      host: {
        id: 1,
        name: `Angelina`,
        isPro: true,
        avatarUrl: `img/avatar-angelina.jpg`
      },
      description: `Discover daily local life in city center.`,
      isPremium: true,
      isFavorite: true,
      location: {
        latitude: 0,
        longitude: 0,
      }
    }
  ],

  cityCoords: {
    cityName: `Cologne`,
    latitude: 0,
    longitude: 0,
    zoom: 13
  },

  offersCoords: [{
    offerId: 1,
    latitude: 0,
    longitude: 0,
    zoom: 13
  }],

  info: {
    // eslint-disable-next-line camelcase
    avatar_url: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg`,
    email: `Oliver.conner@gmail.com`,
    id: 1,
    // eslint-disable-next-line camelcase
    is_pro: false,
    name: `12345678`,
  },
};
