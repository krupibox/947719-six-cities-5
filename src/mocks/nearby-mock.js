export default [
  {
    id: 1,
    city: {
      name: `Paris`,
      location: {
        latitude: 0,
        longitude: 0,
      }
    },
    previewImage: `img/apartment-01.jpg`,
    images: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
    ],
    type: `Apartment`,
    price: 80,
    rating: 3.1,
    title: `Beautiful &amp; luxurious apartment at great location`,
    coordinates: [52.3909553943508, 4.85309666406198],
    isPremium: true,
    location: {
      latitude: 0,
      longitude: 0,
    }
  }
];
