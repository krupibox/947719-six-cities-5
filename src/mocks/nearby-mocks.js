export default [
  {
    id: 91,
    city: `Paris`,
    images: [
      `img/room.jpg`,
    ],
    type: `Private room`,
    price: `80`,
    rating: 3.1,
    name: `Wood and stone place`,
    coordinates: [52.3909553943508, 4.85309666406198],
  },
  {
    id: 92,
    city: `Cologne`,
    images: [`img/apartment-02.jpg`],
    type: `Apartment`,
    price: `132`,
    rating: 1.8,
    name: `Canal View Prinsengracht`,
    coordinates: [52.369553943508, 4.85309666406198],
  },
  {
    id: 93,
    city: `Brussels`,
    images: [`img/apartment-03.jpg`],
    type: `Apartment`,
    price: `180`,
    rating: 3.8,
    name: `Nice, cozy, warm big bed`,
    coordinates: [52.3909553943508, 4.929309666406198],
  }
];
