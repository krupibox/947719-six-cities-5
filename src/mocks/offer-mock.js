export default {
  id: 1,
  city: {
    name: `Paris`,
    location: {
      latitude: 0,
      longitude: 0,
    }
  },
  previewImage: `img/apartment-01.jpg`,
  images: [
    `img/apartment-01.jpg`,
    `img/apartment-02.jpg`,
    `img/apartment-03.jpg`,
  ],
  type: `Apartment`,
  price: 80,
  rating: 3.1,
  title: `Beautiful &amp; luxurious apartment at great location`,
  coordinates: [52.3909553943508, 4.85309666406198],
  host: {
    id: 1,
    name: `Angelina`,
    isPro: true,
    avatarUrl: `img/avatar-angelina.jpg`
  },
  description: `Discover daily local life in city center.`,
  isPremium: true,
  location: {
    latitude: 0,
    longitude: 0,
  }
};
