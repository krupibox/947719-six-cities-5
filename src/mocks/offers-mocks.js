export default [
  {
    id: 1,
    city: `Paris`,
    images: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
    ],
    type: `Apartment`,
    price: `80`,
    rating: 3.1,
    name: `Beautiful &amp; luxurious apartment at great location`,
    coordinates: [52.3909553943508, 4.85309666406198],
    isPremium: true
  },
  {
    id: 2,
    city: `Cologne`,
    images: [`img/room.jpg`],
    type: `Private room`,
    price: `132`,
    rating: 1.8,
    name: `Wood and stone place`,
    coordinates: [52.369553943508, 4.85309666406198],
    isPremium: false
  },
  {
    id: 3,
    city: `Brussels`,
    images: [`img/apartment-02.jpg`],
    type: `Apartment`,
    price: `180`,
    rating: 3.8,
    name: `Canal View Prinsengracht`,
    coordinates: [52.3909553943508, 4.929309666406198],
    isPremium: false
  },
  {
    id: 4,
    city: `Amsterdam`,
    images: [`img/apartment-03.jpg`],
    type: `Apartment`,
    price: `80`,
    rating: 4.5,
    name: `Nice, cozy, warm big bed apartment`,
    coordinates: [52.3809553943508, 4.939309666406198],
    isPremium: false
  },
  {
    id: 5,
    city: `Hamburg`,
    images: [`img/room.jpg`],
    type: `Private room`,
    price: `132`,
    rating: 1.8,
    name: `Wood and stone place`,
    coordinates: [52.369553943508, 4.85309666406198],
    isPremium: true
  },
  {
    id: 6,
    city: `Dusseldorf`,
    images: [`img/apartment-02.jpg`],
    type: `Apartment`,
    price: `180`,
    rating: 3.8,
    name: `Canal View Prinsengracht`,
    coordinates: [52.3909553943508, 4.929309666406198],
    isPremium: false
  },
  {
    id: 7,
    city: `Cologne`,
    images: [`img/apartment-03.jpg`],
    type: `Apartment`,
    price: `80`,
    rating: 4.5,
    name: `Nice, cozy, warm big bed apartment`,
    coordinates: [52.3809553943508, 4.939309666406198],
    isPremium: false
  },
];
