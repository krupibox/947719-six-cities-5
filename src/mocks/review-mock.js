const AVATAR_URL = `https://i.pravatar.cc/150?img`;

export default [
  {
    user: {
      id: 1,
      avatarUrl: `${AVATAR_URL}=1`,
      name: `Max`,
    },
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    date: `2020-11-05T09:55:08.500Z`,
    rating: 4.1,
  },
];
