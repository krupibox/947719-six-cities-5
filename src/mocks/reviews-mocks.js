export default [
  [
    {
      user: {
        id: 1,
        avatar: `../img/avatar-max.jpg`,
        name: `Max`,
        isPro: true,
      },
      comment: {
        id: 1,
        rating: 4.1,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        date: 1569006474313
      }
    },
    {
      user: {
        id: 2,
        avatar: `../img/avatar-max.jpg`,
        name: `Sax`,
        isPro: false,
      },
      comment: {
        id: 2,
        rating: 4.5,
        text: `It is my favorite hotel. I have been staying her for several years and can confirm that service level remains at the same high level.`,
        date: 1571598485610
      }
    },
    {
      user: {
        id: 3,
        avatar: `../img/avatar-max.jpg`,
        name: `Rax`,
        isPro: true,
      },
      comment: {
        id: 3,
        rating: 3.8,
        text: `Emplacement parfait, beaucoup de commerces à proximité, le tout dans un lieu traditionnel et agréable, j'ai adoré mon court séjour et aimerait beaucoup y retourner un jour!`,
        date: 1574276894250
      }
    }
  ],
  [{
    user: {
      id: 4,
      avatar: `../img/avatar-max.jpg`,
      name: `Wax`,
      isPro: true,
    },
    comment: {
      id: 4,
      rating: 3.8,
      text: `Very nice hotel, helpful and friendly staff (thank you guys for everything), good breakfast, clean and comfy room.`,
      date: 1576868903507
    }
  }],
  [{
    user: {
      id: 4,
      avatar: `../img/avatar-max.jpg`,
      name: `Tax`,
      isPro: false,
    },
    comment: {
      id: 4,
      rating: 4.5,
      text: `Breakfast was excellent. Everything was really tasty. Location perfect, in a walking distance from everything. Staff was friendly and welcoming. Hotel was really silent.`,
      date: 1576868903507
    }
  }],
  [{
    user: {
      id: 5,
      avatar: `../img/avatar-max.jpg`,
      name: `Bax`,
      isPro: false,
    },
    comment: {
      id: 5,
      rating: 3.2,
      text: `The only drawback for me was that I could hear the traffic through the window in my room. It was therefore not often i opened the window to get fresh air.`,
      date: 1576868903507
    }
  }],
  [{
    user: {
      id: 6,
      avatar: `../img/avatar-max.jpg`,
      name: `Dax`,
      isPro: true,
    },
    text: {
      id: 6,
      rating: 3.8,
      comment: `Very nice hotel, helpful and friendly staff (thank you guys for everything), good breakfast, clean and comfy room.`,
      date: 1576868903507
    }
  }],
  [{
    user: {
      id: 7,
      avatar: `../img/avatar-max.jpg`,
      name: `Zax`,
      isPro: false,
    },
    comment: {
      id: 7,
      rating: 4.5,
      text: `I stayed here for one night and it was an unpleasant experience.`,
      date: 1576868903507
    }
  }],
  [{
    user: {
      id: 8,
      avatar: `../img/avatar-max.jpg`,
      name: `Pax`,
      isPro: false,
    },
    comment: {
      id: 8,
      rating: 3.2,
      text: `The room was spacious and clean. The pool looked nothing like the photos and desparately needs a clean. The sauna and spa were closed for lunar new year holiday.`,
      date: 1576868903507
    }
  }],
];
