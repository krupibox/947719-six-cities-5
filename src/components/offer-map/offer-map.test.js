import OfferMap from './offer-map';
import {TestMock} from '../../__mocks__/mocks';

describe(`<OfferMap/>`, () => {
  it(`should render correctly`, () => {
    const tree = renderer
      .create(
          <OfferMap
            cityCenterCoords={TestMock.offer.city.location}
            offerCoords={TestMock.offer.coordinates}
            currentCoords={TestMock.offer.city.location}
          />,
          {
            createNodeMock: () => { }
          })
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
