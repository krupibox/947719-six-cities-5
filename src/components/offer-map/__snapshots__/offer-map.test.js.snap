// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OfferMap/> should render correctly 1`] = `
<div
  id="map"
  style={
    Object {
      "height": "100%",
    }
  }
/>
`;
